crypto/x509/x509_vfy.c

```diff
@@ -974,9 +994,11 @@ static int check_revocation(X509_STORE_CTX *ctx)
 {
     int i = 0, last = 0, ok = 0;
 
-    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK) == 0)
+    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK) == 0
+            && (ctx->param->flags & X509_V_FLAG_OCSP_CHECK) == 0)
         return 1;
-    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK_ALL) != 0) {
+    if ((ctx->param->flags & X509_V_FLAG_CRL_CHECK_ALL) == 0
+            || (ctx->param->flags & X509_V_FLAG_OCSP_CHECK_ALL) == 0) {
```

Comment:
Must not change `!= 0` to `== 0` and can be simplified to
`(ctx->param->flags & (X509_V_FLAG_CRL_CHECK_ALL | X509_V_FLAG_OCSP_CHECK_ALL)) != 0`